### ISSUE TITLE
[nswag, v12.0.0.0] SwaggerToTypeScriptClientGenerator ignores the "readOnly" flag for properties

### ISSUE THREAD

**Issue by JechoJekov** [RSuter/NSwag#1763]
The TypeScript code generated by `nswag` does not add the `readonly` TypeScript modifier for properties that has `readOnly` set to `true` in the Swagger specification file.

According to the `Class.liquid` template the modifier is added if `property.IsReadOnly` is `true`. I tried to debug using a custom template and `property.IsReadOnly` is set to `false` even for read-only properties.

I have observed this behavior for quite some time now and initially thought this could be by design. However, it seems to me that this is actually a bug.

**Comment by RSuter** [comment]
"readOnly" is only defined for OpenAPI v3, for Swagger 2.0 you need to use "x-readOnly"

**Comment by JechoJekov** [comment]
@RSuter Thank you very much for your response. Is it possible to set this value by using attributes or otherwise, how can I set it for a property?

**Comment by RSuter** [comment]
Isnt there a ReadOnlyAttribute?

**Comment by RSuter** [comment]
Yep, this one: https://github.com/RSuter/NJsonSchema/wiki/JsonSchemaGenerator#read-only-property

**Comment by RSuter** [comment]
I think it even works with an own attribute with the same name but in your own namespace

**Comment by JechoJekov** [comment]
I am using this attribute on all read-only properties and it does set the "readOnly" flag in the schema. However, it seems that the TypeScript generator ignores this flag,

**Comment by RSuter** [comment]
The TypeScriptVersion must be >= 2.0: 

https://github.com/RSuter/NJsonSchema/blob/master/src/NJsonSchema.CodeGeneration.TypeScript/Models/PropertyModel.cs#L111

**Comment by JechoJekov** [comment]
I use TypeScript 2.7:

```Json
"codeGenerators": {
    "swaggerToTypeScriptClient": {
        "typeScriptVersion": 2.7
    }
}
```

**Comment by RSuter** [comment]
Strange, for classes it should use the linked IsReadOnly model property: 

https://github.com/RSuter/NJsonSchema/blob/master/src/NJsonSchema.CodeGeneration.TypeScript/Templates/Class.liquid#L9

For interfaces: 

https://github.com/RSuter/NJsonSchema/blob/master/src/NJsonSchema.CodeGeneration.TypeScript/Templates/Interface.liquid#L9

**Comment by RSuter** [comment]
Are you using OpenAPI 3.0 schema?

**Comment by JechoJekov** [comment]
I tried `NJsonSchema.SchemaType.JsonSchema` and Swagger UI displays the following message:

```
Unable to render this definition

swagger and openapi fields cannot be present in the same Swagger or OpenAPI definition. Please remove one of the fields.

Supported version fields are swagger: "2.0" and those that match  openapi: 3.0.n (for example, openapi: 3.0.0).
```

![swagger-error](https://user-images.githubusercontent.com/3956646/49114783-b9766280-f2a1-11e8-8b47-405cacfe75ca.jpg)

The JSON schema contains both `swagger` and `openapi` fields:

```js
{
  "x-generator": "NSwag v12.0.0.0 (NJsonSchema v9.12.2.0 (Newtonsoft.Json v11.0.0.0))",
  "swagger": "2.0",
  "openapi": "3.0.0",
}
```

Perhaps this is a different issue.

**Comment by RSuter** [comment]
`SchemaType.JsonSchema` is not allowed for SwaggerDocuments - but because the base of NSwag is NJS this cannot be avoided. You have to change the type in the generator settings and not on the document as it will not change the schema structure according to the type - it will only change the semantics how it is processed (e.g. null handling, etc)

**Comment by RSuter** [comment]
The easiest way would be to clone the NJsonSchema repo, create a unit test for your case and check whether this works correctly... or is there already a test?
